File: csharp/Services/PlayerService.cs
using Microsoft.Extensions.Logging;
using NewNanManager.Client.Http;
using NewNanManager.Client.Models;

namespace NewNanManager.Client.Services;

/// <summary>
/// 玩家管理服务
/// </summary>
public class PlayerService : HttpClientBase
{
    public PlayerService(HttpClient httpClient, ILogger? logger = null) : base(httpClient, logger)
    {
    }

    /// <summary>
    /// 获取玩家列表
    /// </summary>
    /// <param name="page">页码</param>
    /// <param name="pageSize">每页大小</param>
    /// <param name="search">搜索关键词</param>
    /// <param name="townId">城镇ID</param>
    /// <param name="banMode">封禁模式</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>玩家列表数据</returns>
    public async Task<PlayersListData> ListPlayersAsync(
        int? page = null,
        int? pageSize = null,
        string? search = null,
        int? townId = null,
        BanMode? banMode = null,
        CancellationToken cancellationToken = default)
    {
        var queryParams = new Dictionary<string, object?>
        {
            ["page"] = page,
            ["page_size"] = pageSize,
            ["search"] = search,
            ["town_id"] = townId,
            ["ban_mode"] = banMode.HasValue ? (int)banMode.Value : null
        };

        var queryString = BuildQueryString(queryParams);
        return await GetAsync<PlayersListData>($"/api/v1/players{queryString}", cancellationToken);
    }

    /// <summary>
    /// 创建玩家
    /// </summary>
    /// <param name="request">创建玩家请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>创建的玩家信息</returns>
    public async Task<Player> CreatePlayerAsync(CreatePlayerRequest request, CancellationToken cancellationToken = default)
    {
        return await PostAsync<Player>("/api/v1/players", request, cancellationToken);
    }

    /// <summary>
    /// 验证玩家登录
    /// </summary>
    /// <param name="request">登录验证请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>验证结果</returns>
    public async Task<ValidateLoginData> ValidateLoginAsync(ValidateLoginRequest request, CancellationToken cancellationToken = default)
    {
        return await PostAsync<ValidateLoginData>("/api/v1/players/validate-login", request, cancellationToken);
    }

    /// <summary>
    /// 获取玩家详情
    /// </summary>
    /// <param name="id">玩家ID</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>玩家信息</returns>
    public async Task<Player> GetPlayerAsync(int id, CancellationToken cancellationToken = default)
    {
        return await GetAsync<Player>($"/api/v1/players/{id}", cancellationToken);
    }

    /// <summary>
    /// 更新玩家信息
    /// </summary>
    /// <param name="id">玩家ID</param>
    /// <param name="request">更新请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>更新后的玩家信息</returns>
    public async Task<Player> UpdatePlayerAsync(int id, UpdatePlayerRequest request, CancellationToken cancellationToken = default)
    {
        return await PutAsync<Player>($"/api/v1/players/{id}", request, cancellationToken);
    }

    /// <summary>
    /// 删除玩家
    /// </summary>
    /// <param name="id">玩家ID</param>
    /// <param name="cancellationToken">取消令牌</param>
    public async Task DeletePlayerAsync(int id, CancellationToken cancellationToken = default)
    {
        await DeleteAsync($"/api/v1/players/{id}", cancellationToken);
    }

    /// <summary>
    /// 封禁玩家
    /// </summary>
    /// <param name="playerId">玩家ID</param>
    /// <param name="request">封禁请求</param>
    /// <param name="cancellationToken">取消令牌</param>
    public async Task BanPlayerAsync(int playerId, BanPlayerRequest request, CancellationToken cancellationToken = default)
    {
        await PostAsync($"/api/v1/players/{playerId}/ban", request, cancellationToken);
    }

    /// <summary>
    /// 解封玩家
    /// </summary>
    /// <param name="playerId">玩家ID</param>
    /// <param name="cancellationToken">取消令牌</param>
    public async Task UnbanPlayerAsync(int playerId, CancellationToken cancellationToken = default)
    {
        await PostAsync($"/api/v1/players/{playerId}/unban", cancellationToken: cancellationToken);
    }
}

File: kotlin/src/main/kotlin/com/nanmanager/client/services/PlayerService.kt
package com.nanmanager.client.services

import com.nanmanager.client.models.*
import io.ktor.client.*

/**
 * 玩家管理服务模块
 * 基于模块化架构的标准化API定义
 */
class PlayerService(
    httpClient: HttpClient,
    baseUrl: String,
    token: String
) : BaseService(httpClient, baseUrl, token) {

    /**
     * 创建玩家
     */
    suspend fun createPlayer(request: CreatePlayerRequest): Player {
        return post("/api/v1/players", request)
    }

    /**
     * 获取玩家详情
     */
    suspend fun getPlayer(id: Int): Player {
        return get("/api/v1/players/$id")
    }

    /**
     * 更新玩家信息
     */
    suspend fun updatePlayer(id: Int, request: UpdatePlayerRequest): Player {
        return put("/api/v1/players/$id", request)
    }

    /**
     * 删除玩家
     */
    suspend fun deletePlayer(id: Int) {
        delete("/api/v1/players/$id")
    }

    /**
     * 获取玩家列表
     */
    suspend fun listPlayers(request: ListPlayersRequest = ListPlayersRequest()): ListPlayersResponse {
        val params = buildParams(mapOf(
            "page" to request.page,
            "page_size" to request.pageSize,
            "search" to request.search,
            "town_id" to request.townId,
            "ban_mode" to request.banMode?.value,
            "name" to request.name,
            "qq" to request.qq,
            "qqguild" to request.qqguild,
            "discord" to request.discord
        ))
        return get("/api/v1/players", params)
    }

    /**
     * 封禁玩家
     */
    suspend fun banPlayer(playerId: Int, request: BanPlayerRequest) {
        val body = mapOf(
            "ban_mode" to request.banMode,
            "duration_seconds" to request.durationSeconds,
            "reason" to request.reason
        )
        post<EmptyResponse>("/api/v1/players/$playerId/ban", body)
    }

    /**
     * 解封玩家
     */
    suspend fun unbanPlayer(playerId: Int) {
        post<EmptyResponse>("/api/v1/players/$playerId/unban", emptyMap<String, Any>())
    }

    /**
     * 批量玩家验证
     */
    suspend fun validate(request: ValidateRequest): ValidateResponse {
        val body = mapOf(
            "players" to request.players.map { player ->
                mapOf(
                    "player_name" to player.playerName,
                    "ip" to player.ip,
                    "client_version" to player.clientVersion,
                    "protocol_version" to player.protocolVersion
                )
            },
            "server_id" to request.serverId,
            "login" to request.login,
            "timestamp" to (if (request.timestamp == 0L) System.currentTimeMillis() else request.timestamp)
        )
        return post("/api/v1/players/validate", body)
    }

    /**
     * 验证玩家登录（向后兼容）
     * @deprecated 请使用 validate 方法进行批量验证
     */
    @Deprecated("请使用 validate 方法进行批量验证")
    suspend fun validateLogin(request: ValidateLoginRequest): ValidateLoginResponse {
        return post("/api/v1/players/validate-login", request)
    }
}
